deephyper/search/hps/_cbo.py

```diff
@@ -244,10 +262,12 @@ def _search(self, max_evals, timeout):
                 opt_y = []
                 for cfg, obj in new_results:
                     x = list(cfg.values())
-                    if np.isreal(obj):
+                    if np.all(np.isreal(obj)):
                         opt_X.append(x)
-                        opt_y.append(-obj)  #! maximizing
-                    elif type(obj) is str and "F" == obj[0]:
+                        opt_y.append(np.negative(obj).tolist())  #! maximizing
```

Comment:
is single objective optimization still working with this? each objective is now a list (of length 1 for single-objective)

Reply:
According to the documentation [here](https://numpy.org/doc/stable/reference/generated/numpy.ndarray.tolist.html), it will not be a list but a simple Python scalar.